x/leverage/types/keys.go

```diff
@@ -172,3 +172,30 @@ func CreateLendAPYKey(tokenDenom string) []byte {
 	key = append(key, []byte(tokenDenom)...)
 	return append(key, 0) // append 0 for null-termination
 }
+
+// GetAddressFromKeyWithPrefix the key should be exactly:
+// prefix | lengthPrefixed(addr)
+func GetAddressFromKeyWithPrefix(key []byte, prefix []byte) sdk.AccAddress {
+	// key is prefix | lengthPrefixed(addr) | ....
+	prefixLength := len(prefix)
+	lengthAddr := int(key[prefixLength])
+	// plus 1 because of the byte that represent the size of lengthPrefixedAddress
+	initAddr := prefixLength + 1
+	endAddr := initAddr + lengthAddr
+
+	return key[initAddr:endAddr]
+}
```

Comment:
Good idea turning this pattern into a function.
```suggestion
// AddressFromKey extracts address from a key of the form
// prefix | lengthPrefixed(addr) | ...
func AddressFromKey(key []byte, prefix []byte) sdk.AccAddress {
	addrLength := int(key[prefixLength])
	return key[len(prefix)+1:len(prefix)+1+addrLength]
}
```
- comment and function simplify
- renamed function (we don't have to mention prefix)